vignettes/articles/modify-functions.Rmd

```diff
@@ -0,0 +1,197 @@
+- `modify_footnote_spanning_header()`/`remove_footnote_spanning_header()`: Update/remove footnotes corresponding to a table spanning header
+
+These functions are used to modify table footnotes. Each function corresponds to footnotes relating to a specific part of the table - the table body (including labels), header, and spanning header(s).
+
+Footnotes will be listed (and numbered) in the order in which they appear in the table, from top-left to bottom-right. Duplicate footnotes will only be numbered and displayed once.
+
+By default any pre-existing footnotes will be overwritten and replaced with the newly specified footnotes. This can be disabled so that previous footnotes are retained by setting `replace = FALSE` in each of the footnote modifier functions.
+
+The `remove_*()` functions will remove all footnotes for the associated table part by default. See the function documentation for custom footnote removal options.
+
+```{r foot}
+x |>
+  # add footnote associated with the grade label within the label column
+  modify_footnote_body(
+    footnote = "Tumor grade was assessed _before_ treatment began",
+    columns = "label",
+    rows = variable == "grade" & row_type == "label"
+  ) |>
+  # add footnote associated with the grade III p-value cell
+  modify_footnote_body(
+    footnote = "Reported p-value outside of the range of interest",
+    columns = p.value,
+    rows = variable == "grade" & label == "III"
+  ) |>
+  # add footnote associated with the header over all columns
+  modify_footnote_header(
+    footnote = "All subjects received treatment",
+    columns = everything(),
+    replace = FALSE
+  )
+```
+
+## Source Note & Caption Modifiers
+
+- `modify_abbreviation()`/`remove_abbreviation()`: Update/remove table text abbreviations source note
+- `modify_caption()`: Update table caption
+- `modify_source_note()`/`remove_source_note()`: Update/remove table source notes
+
+These functions are used to modify table source notes and captions. Source notes are similar to footnotes but are not linked to a specific cell in the table.
+
+All abbreviations supplied via `modify_source_note()` are coalesced into a single source note.
+
+Source notes are displayed after footnotes, with the abbreviations source note listed before any other source notes.
+
+The formatting and appearance of source notes and captions are largely dependent on the table engine used, as some features are not supported by all table engines. See the individual function documentation for details.
+
+```{r source}
+x |>
+  modify_source_note("Results as of June 26, 2015") |>
+  modify_abbreviation("I = Grade 1") |>
+  remove_abbreviation("OR = Odds Ratio")
+```
+
+## Advanced Modifiers
+
+**These functions should be used by developers or advanced users only.**
+
+- `modify_table_body()`: Update `x$table_body` directly 
+- `modify_table_styling()`: Update `x$table_styling` directly 
+- `modify_column_merge()`/`remove_column_merge()`: Merge/unmerge two or more columns
```

Comment:
Maybe this function should be in its own section? I think users will be using this for listing often, so I don't want to give the message they should not use it. Take a look at the note in the help file, and we can be clear that it should be used in simple situations (like listings).

Reply:
Done